ui/packages/catalog-ui-search/src/main/webapp/component/query-basic/query-basic.view.js

```diff
@@ -45,17 +48,44 @@ function getMatchTypeAttribute() {
     : 'datatype'
 }
 
-function isTypeLimiter(filter) {
-  let typesFound = {}
-  filter.filters.forEach(subfilter => {
-    typesFound[CQLUtils.getProperty(subfilter)] = true
+const getMatchTypes = memoize(async () => {
+  const matchTypeAttr = getMatchTypeAttribute()
+  const json = await query({
+    count: 0,
+    cql: "anyText ILIKE '*'",
+    facets: [matchTypeAttr],
   })
-  typesFound = Object.keys(typesFound)
-  return (
-    typesFound.length === 2 &&
-    typesFound.indexOf('metadata-content-type') >= 0 &&
-    typesFound.indexOf(getMatchTypeAttribute()) >= 0
-  )
+  const facets = json.facets[matchTypeAttr] || []
+  return facets
+    .map(facet => facet.value)
```

Comment:
:question: Should we still use the metacard type enums if the facet result is empty? Or maybe merge the two? I'm thinking about cases where sources don't support faceted queries but still have data with known values

Reply:
I think it makes sense to fall back to the enums if there are no faceted results.

Reply:
Updated.